src/main/java/org/datadog/jmxfetch/reporter/Reporter.java

```diff
@@ -119,7 +119,15 @@ public void sendMetrics(List<Metric> metrics, String instanceName, boolean canon
                 }
                 sendMetricPoint(metricType, metricName, delta, tags);
 
-            } else { // The metric should be 'counter'
+            } else {
+                // `counter` and `rate` are equivalent and both accepted as valid.
+                // We continue accepting other cases for backward compatibility.
+                if (!"counter".equals(metricType) && !"rate".equals(metricType)) {
+                    log.warn("Invalid metric type " + metricType + " for metric " + metricName
+                            + ". The metric will be processed as rate"
+                            + " (this is a DEPRECATED behaviour, use a valid type instead).");
+                }
```

Comment:
Note: I tried to have a logic like

```
            if ("gauge".equals(metricType) || "histogram".equals(metricType)) {
                ...
            } else if ("monotonic_count".equals(metricType)) {
                ...
            } else if ("counter".equals(metricType) || "rate".equals(metricType)) {
                sendCounter(...)
            } else {
                // issue warning and deprecation
               sendCounter(...)
            }
```

But the issue is that we will need to pass about 7 args to sendCounter, which is not great. That's why I went to this low changelog approach you see here that only add the warning.